BUILDING.md

```diff
@@ -53,66 +29,169 @@ following packages are installed:
 > are attempting to use the ConanCenter cpython package as an
 > alternative, we have encountered missing header issues that may
 > require additional work/configuration to resolve.
 
-## Building with CMake
-
-We use the CMake build system for compiling the C++ core library and
-its Python bindings.
-
-By default, the project will build shared libraries and set the
-installation directory to `dist` under the build directory. This can be
-overridden by setting the built-in CMake variables `BUILD_SHARED_LIBS`
-and `CMAKE_INSTALL_PREFIX`, respectively.
+## Building
 
-If shared library builds are disabled, then the core library will be
-built and installed as a static library, and statically linked into the
-Python module.
+OpenAssetIO consists of a core C++ project, as well as a set of C and
+Python bindings to that core libary. These modules are organized into a
+CMake project, and are all built together by default.
 
-In order to build and install the binary artifacts and Python sources,
-assuming the current working directory is at the repository root, run
+Assuming your build environment has the required dependencies,
+[(See Above)](#build-dependencies), OpenAssetIO can be built and installed by
+running :
 
 ```shell
 cmake -S . -B build
 cmake --build build
 cmake --install build
 ```
 
-The generated install tree can then be used directly (e.g. by extending
+> **Note**
+>
+> A common error to encounter is to find that CMake cannot resolve
+> its dependencies via [`find_package`](https://cmake.org/cmake/help/latest/command/find_package.html)
+>
+> A convenient way to solve this is to use the [`CMAKE_TOOLCHAIN_FILE`](https://cmake.org/cmake/help/v3.24/envvar/CMAKE_TOOLCHAIN_FILE.html)
+> environment variable to augment CMake's package search paths.
+>
+> For example, to use [`conan`](https://docs.conan.io/en/1.46/installation.html)
+> to resolve dependencies, (as we do during internal development) run
+> the following commands from the root of the repository prior to
+> building :
+>
+> ```shell
+> conan install -if .conan resources/build --build=missing
+> export CMAKE_TOOLCHAIN_FILE=`pwd`/.conan/conan_paths.cmake
+> ```
+>
+> (These commands assume a Linux host, translate as necessary).
+>
+> This installs OpenAssetIO's dependencies as described in
+> `/resources/build/conanfile.py`, and allows CMake to deduce the
+> locations via referencing the generated `conan_paths.cmake`.
+
+The modules are installed to `/build/dist`. This default location can
+be overridden by setting `CMAKE_INSTALL_PREFIX` prior to build.
+
+By default, OpenAssetIO builds as a shared library. This can be
+overriden by setting `BUILD_SHARED_LIBS` prior to build.
+If shared library builds are disabled, then the core library will
+be built and installed as a static library, and statically linked into
+the Python/C modules.
+
+In installing the python module, OpenAssetIO creates a valid python
+install tree, placed at `build/dist/{python-version}/site-packages`.
```

Comment:
[consider] Not quite convinced on the wording here.  I think a "valid Python install tree" isn't really a thing - you can extend `PYTHONPATH` into any directory you want.  Oh I think you mean a Python package?  Yeah perhaps best to use the word "package"
```suggestion
package, placed at `build/dist/lib/{python-version}/site-packages/openassetio`.
```
It's also obfuscating the C++ component. The C/C++ libraries is what the reference to `LD_LIBRARY_PATH` was originally about, and the "further packaging step" was also originally about packaging along with the non-Python artifacts (e.g. Conan, Rez, tar.gz).

Reply:
Yeah, still unclear on this terminology. I thought the term "install tree" meant like, we generated an output structure that is identical to the structure expected in python-land when libraries are installed.

Nonetheless, will change to your phrasing as it's obviously correct.

Reply:
It's still correct to follow on with ... "the generated install tree can then be used directly..."? that comes from the original docs.

Reply:
I think it's "install tree" that is confusing in a python context.  It would be "the generated python package can then be..."

Reply:
[bd3a6db](https://github.com/OpenAssetIO/OpenAssetIO/pull/711/commits/bd3a6db46c6ca142d77b71699f7c470756f4f218)

Reply:
The rewrite is OK in and of itself, but I think we've lost some of the original intention of that paragraph. It's now very Python-centric, when the original intention was that the install tree as a whole could be used e.g.
* Directly in a C++ application, by extending `LD_LIBRARY_PATH` (or `PATH` on Windows).
* Directly in a Python application, by extending `PYTHONPATH` (plus probably `PATH` on Windows).
* For consumption in a CMake project using `find_package`, by extending `CMAKE_PREFIX_PATH` (or equivalent).
* For archiving as a zip file for redistribution (since all RPATHS are relative, so it's theoretically relocatable).
* For consumption by an packaging tool such as Conan, Rez, vcpkg, for further redistribution.

Admittedly what was there before was rather terse to cover all these options.

Reply:
Pretty terse indeed. I'm rather not sure what to do here. Now that we've listed the output formulations of both the C++ and python libs, i'm somewhat tempted to just delete the

> This generated python package can then be used directly (e.g. by
extending `PYTHONPATH` and/or `LD_LIBRARY_PATH`), or as the source for a
further packaging step.

fragment that is causing all this consternation, is letting people know they can use the output dirs directly helpful at all?

Reply:
Also includes a couple of spelling fixes [2f0ec37](https://github.com/OpenAssetIO/OpenAssetIO/pull/711/commits/2f0ec37ce2bd7579a18f82aaa581c32c1ff877b7)